src/Build.UnitTests/BackEnd/BuildRequestEngine_Tests.cs

```diff
@@ -342,9 +342,15 @@ public void TestEngineShutdownWhileActive()
 
             VerifyEngineStatus(BuildRequestEngineStatus.Uninitialized);
             _engine.InitializeForBuild(new NodeLoggingContext(_host.LoggingService, 0, false));
+            // We neeed to get the status changed AutoResetEvent returned to the non-signaled state correctly for verifying the engine status via waiting for a signal next time.
+            // Make sure the status change finishes
+            VerifyEngineStatus(BuildRequestEngineStatus.Idle);
```

Comment:
The added `VerifyEngineStatus` for the additional states looks good to me (here and all the others)!

It's true that it's actually very similar to extending the sleep (as mentioned by @Forgind) - however it's more fail-fast and succeed-fast then just sleep.